architecture/COMPONENTS.md

```
@@ -254,6 +252,71 @@ export function Button({ children, ...props }: ButtonProps) {
+- Critical paths: auth, checkout, onboarding
+- Dynamic content: list items, cards (`data-testid="link-item-{id}"`)
+- Conditional UI: elements that appear/disappear
+- E2E smoke test entry points
+
+### When to Skip
+- Semantic HTML: use `getByRole('button')` instead
+- Static content: use `getByText('Welcome')`
+- Elements with clear accessibility selectors
+
+### Naming Convention
+```
```

markdownlint-cli2 (0.18.1)

Fenced code blocks should have a language specified

(MD040, fenced-code-language)